Adopt `PluginDirectoryServiceRemote` changes

crazytonyli opened the pull request:
> [!Note]
> This PR is built on top of #22612.

This PR updates the app code to accept the `PluginDirectoryServiceRemote` breaking changes introduced in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/725.

## Test Instructions

1. Log into a self hosted site
2. Go to "Plugins"
3. Verify the Plugins screen can be loaded
4. Verify plugin details can be viewed.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the "Test Instructions"

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist: N/A

crazytonyli committed:
Adopt to `PluginDirectoryServiceRemote` changes

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio reviewed (approved):
Approving to unblock while waiting for the Kit reference to be updated to the merge commit of the PR into `trunk`.


I installed the Jetpack App Center build and verified plugins worked.

<table>
<tr>
 <td>

![IMG_A1741D133F8B-1](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/14bb020c-2038-4c2f-bff0-ee588e281b91)

</td>
 <td>

![IMG_A1741D133F8B-2](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/fdb98640-dcd2-433e-9f1a-0e91b89be570)

</td>
</table>

crazytonyli committed:
Update WordPressKit

crazytonyli committed:
Add a release note

mokagio committed:
Use a WordPressKit version that should fix the Reader loading issues

See how the UI tests failed in
https://buildkite.com/automattic/wordpress-ios/builds/20452#018dbe88-2fc5-4998-ad35-1b455f547b59

The WordPressKit PR is https://github.com/wordpress-mobile/WordPressKit-iOS/pull/727

mokagio commented on Podfile:
```diff
@@ -51,7 +51,7 @@ end
 def wordpress_kit
   # pod 'WordPressKit', '~> 13.0'
-  pod 'WordPressKit', git: 'https://github.com/wordpress-mobile/WordPressKit-iOS.git', commit: 'dfb2134ca3b91233d44b3551d91844bf2087996f'
+  pod 'WordPressKit', git: 'https://github.com/wordpress-mobile/WordPressKit-iOS.git', commit: '3df47e113c7647b76c049180e39489e3af7a27dc'
```
This commit points to https://github.com/wordpress-mobile/WordPressKit-iOS/pull/727 which, I'm happy to say, seems to address the reader issue that the UI tests picked up in previous runs (see #22612)

crazytonyli commented:
Changes in this PR have been moved to https://github.com/wordpress-mobile/WordPress-iOS/pull/22612
